File: client-packs/frontend-72-route-equivalence-v2/formal-campaign/artifacts/engine/profiles/flutter/project/lib/elmos_bounded_navigation.dart
// Direct identity projection for bounded-navigation-v1 compatibility; no second route literal is permitted.
import 'elmos_bounded_interaction.dart';

typedef ElmosBoundedRoute = Map<String, Object?>;
extension ElmosBoundedRouteFields on ElmosBoundedRoute {
  String get id => this['id']! as String;
  String get path => this['path']! as String;
  String get title => this['title']! as String;
  String get text => this['text']! as String;
  bool get requiresAuth => this['requiresAuth']! as bool;
  bool get deepLink => this['deepLink']! as bool;
}
final Map<String, Object?> elmosBoundedNavigation = <String, Object?>{
  'routes': elmosFrontendInteractionNavigation['routes']!,
};
final List<Object?> elmosBoundedRoutes = elmosBoundedNavigation['routes']! as List<Object?>;
ElmosBoundedRoute elmosRoute(Object? raw) => raw! as ElmosBoundedRoute;
ElmosBoundedRoute get elmosFirstRoute => elmosRoute(elmosBoundedRoutes.first);
ElmosBoundedRoute elmosSelectBoundedRoute(String path) {
  if (elmosBoundedRoutes.isEmpty) throw StateError('bounded navigation requires at least one route');
  return elmosRoute(elmosBoundedRoutes.firstWhere((raw) => elmosRoute(raw).path == path, orElse: () => elmosBoundedRoutes.first));
}

File: client-packs/frontend-72-route-equivalence-v2/formal-campaign/artifacts/engine/profiles/react-native/project/src/elmos-bounded-navigation.ts
// Direct identity projection for bounded-navigation-v1 compatibility; no second route literal is permitted.
import { ELMOS_FRONTEND_INTERACTION } from "./elmos-bounded-interaction";
export const ELMOS_BOUNDED_NAVIGATION = {
  schemaVersion: "1.0", profile: "bounded-navigation-v1",
  projectTitle: ELMOS_FRONTEND_INTERACTION.projectTitle,
  navigation: { label: ELMOS_FRONTEND_INTERACTION.navigation.label },
  render: { mainRole: ELMOS_FRONTEND_INTERACTION.accessibilityFocus.mainRole, headingLevel: ELMOS_FRONTEND_INTERACTION.accessibilityFocus.headingLevel },
  fallback: { strategy: ELMOS_FRONTEND_INTERACTION.navigation.fallback },
  routes: ELMOS_FRONTEND_INTERACTION.navigation.routes,
} as const;
export const ELMOS_ROUTES = ELMOS_BOUNDED_NAVIGATION.routes;

export function elmosSelectBoundedRoute(path: string) {
  const selected = ELMOS_ROUTES.find(route => route.path === path);
  const fallback = ELMOS_ROUTES[0];
  if (!fallback) throw new Error("bounded navigation requires at least one route");
  return selected ?? fallback;
}
